### PR TITLE
Dummy task

### PR DESCRIPTION
**Pull request by azawlocki**

### CODE REVIEW

File: golem/model.py [hunk 1]
```diff
@@ -13,14 +15,18 @@ def initialize_connection(self, conn):
         self.execute_sql('PRAGMA busy_timeout = 30000')
 
 
-db = SqliteFKTimeoutDatabase(DATABASE_NAME, threadlocals=True)
+db = SqliteFKTimeoutDatabase(None, threadlocals=True)
 
 
 class Database:
-    def __init__(self):
+    def __init__(self, name=None):
```
Comment (chfast) [inline golem/model.py]: Can't it be just `name=DATABASE_NAME` (hint: Python's strings are immutable)?

Comment (azawlocki) [inline golem/model.py]: Note that changes related to databases are merged from 0d2ec0810611abd05ee3f5d96349a498034782d2 (I needed separate databases to safely run multiple nodes, each in a separate process).

File: tests/golem/task/dummy/computation.py [hunk 2]
```diff
@@ -26,7 +26,9 @@ def find_pow(input_data, difficulty, result_size):
     :param int result_size:
     :rtype long:
     """
-    solution = random.getrandbits(result_size * 4)
+    num_bits = result_size * 4
+    # This ensures that the generated number will not start with 0's as hex
+    solution = random.getrandbits(num_bits - 1) + (1 << (num_bits - 1))
```
Comment (chfast) [inline tests/golem/task/dummy/computation.py]: I would use binary or operator `|` to make it easier to follow.

File: tests/golem/task/dummy/runner.py [hunk 3]
```diff
@@ -1,47 +1,208 @@
...
-            logger.info('Dummy task still not finished')
-        time.sleep(10)
-    # how to stop the client gracefully?
-    logger.info('Stopping the client...')
-    client.stop_network()
-    reactor.stop()
-    logger.info('Bye')
-    sys.exit(0)
+    def report(msg):
+        print "[REQUESTING NODE {}] {}".format(os.getpid(), msg)
```
Comment (chfast) [inline tests/golem/task/dummy/runner.py]: Can you align PID number?

Comment (azawlocki) [inline tests/golem/task/dummy/runner.py]: I can, but why bother?

Comment (chfast) [inline tests/golem/task/dummy/runner.py]: Just to make it look nicer as PID typically can be 4 or 5 digit long.

File: tests/golem/task/dummy/runner.py [hunk 4]
```diff
@@ -1,47 +1,208 @@
...
 
-import logging.config
-from os import path
-import thread
+import os
+import re
+import subprocess
+import sys
+from threading import Thread
+import time
```
Comment (chfast) [inline tests/golem/task/dummy/runner.py:10]: I've found a nice way to sort imports: first `import`s then `from`s. More in PEP8: https://www.python.org/dev/peps/pep-0008/#id20.

File: tests/golem/task/dummy/runner.py [hunk 5]
```diff
@@ -1,47 +1,208 @@
...
 
-client = start_client()
+REQUESTING_NODE_KIND = "requester"
+COMPUTING_NODE_KIND = "computer"
 
-params = DummyTaskParameters(1024, 2048, 256, 0x0001ffff)
-task = DummyTask(client.get_id(), params, 3)
-client.enqueue_new_task(task)
 
+def run_requesting_node(num_subtasks = 3):
```
Comment (chfast) [inline tests/golem/task/dummy/runner.py]: No space around `=`.

File: tests/golem/task/dummy/runner.py [hunk 6]
```diff
@@ -1,47 +1,208 @@
...
+        print "[COMPUTING NODE {}] {}".format(os.getpid(), msg)
+
+    start_time = time.time()
+    report("Starting...")
+    client = start_client()
+    report("Started in {:.1f} s".format(time.time() - start_time))
+
+    class DummyEnvironment(Environment):
+        @classmethod
+        def get_id(cls):
```
Comment (chfast) [inline tests/golem/task/dummy/runner.py:79]: If `cls` is not used it can be `@staticmethod`.

File: tests/golem/task/dummy/runner.py [hunk 7]
```diff
@@ -1,47 +1,208 @@
...
+
+    dummy_env = DummyEnvironment()
+    dummy_env.accept_tasks = True
+    client.environments_manager.add_environment(dummy_env)
+
+    report("Connecting to requester node at {}:{} ..."
+           .format(peer_address.address, peer_address.port))
+    client.connect(peer_address)
+
+    def report_status(fail_after = None):
```
Comment (chfast) [inline tests/golem/task/dummy/runner.py]: Style: remove spaces around `=`.

File: tests/golem/task/dummy/runner.py [hunk 8]
```diff
@@ -1,47 +1,208 @@
...
+        t0 = time.time()
+        n = 0
+        while True:
+            if fail_after and time.time() - t0 > fail_after:
+                report("Failure!")
+                reactor.callFromThread(reactor.stop)
+                time.sleep(5)
```
Comment (chfast) [inline tests/golem/task/dummy/runner.py]: In general, there are a lots of long sleeps around. Can they be lowered to run tests/simulations faster?

File: tests/golem/task/dummy/runner.py [hunk 9]
```diff
@@ -1,47 +1,208 @@
...
+    reactor.callInThread(report_status, fail_after)
+    reactor.run()
+    return client  # Used in tests, with mocked reactor
+
+
+# Global var set by a thread monitoring the status of the requester node
+task_finished = False
+
+
+def run_simulation(num_computing_nodes = 2, num_subtasks = 3, timeout = 120,
```
Comment (chfast) [inline tests/golem/task/dummy/runner.py]: Style around default argument values.

File: tests/golem/task/dummy/computation.py (hunk 2 above)
Comment (azawlocki) [inline tests/golem/task/dummy/computation.py]: Done

File: tests/golem/task/dummy/runner.py (hunk 3 above)
Comment (azawlocki) [inline tests/golem/task/dummy/runner.py]: Done

File: tests/golem/task/dummy/runner.py (hunk 4 above)
Comment (azawlocki) [inline tests/golem/task/dummy/runner.py:10]: Done, but I'm not sure PEP 8 says from's should come after import's.

File: tests/golem/task/dummy/runner.py (hunk 5 above)
Comment (azawlocki) [inline tests/golem/task/dummy/runner.py]: Fixed (and in other places as well).

File: tests/golem/task/dummy/runner.py (hunk 6 above)
Comment (azawlocki) [inline tests/golem/task/dummy/runner.py:79]: This is an override of a superclass classmethod.

File: tests/golem/task/dummy/runner.py (hunk 8 above)
Comment (azawlocki) [inline tests/golem/task/dummy/runner.py]: I've removed one sleep() that made a difference.

**Comment by chfast** [comment]
Ok!

**Pull request merged**